### PR TITLE
Enhanced `multiply` of transform `math` type to support `float64` 

### LINKED ISSUE #3059: Enhance `multiply` of transform `math` type with support of `float64`
### What problem are you facing?

Currently  `multiply` of transform `math` type is limited to work with `int` and `int64` only.

If we try the constructs like

```
    - fromFieldPath: "status.maxConnPerSku"
      toFieldPath: "spec.forProvider.criteria[0].threshold"
      transforms:
      - type: convert
        convert:
          toType: float64
      - type: math
        math:
          multiply: 90
```

we will encounter patching failure of
```
 "reason": "ComposeResources", "message": "cannot render composed resource from resource template at index 3: cannot apply the patch at index 4: transform at index 1 returned error: math transform could not resolve: input is required to be a number for math transformer"}
```

In other words, `multiply` does not work with decimals on both sides of the equation: both inputs and multiplication factors are required to be integer only. 

It is seriously limiting in cases we need to deal with decimal values coming from source fields and/or multiply by a non-integer value like e.g. `1.05` or `0.9`

### How could Crossplane help solve your problem?

Enable `multiply` to support `float64`

Related code: https://github.com/crossplane/crossplane/blob/b49ddde2627ac732c53cae4a683024253cfe81b0/apis/apiextensions/v1/composition_transforms.go#L121-L137

### PR DESCRIPTION
**Pull request by Vrashabh-Sontakke**
### Description of your changes

before: `multiply` of transform `math` type is limited to work with `int` and `int64` only.
after: Enabled `multiply` to support `float64`

Fixes #3059 

I have:

- [ x] Read and followed Crossplane's [contribution process].

### CODE REVIEW

File: apis/apiextensions/v1/composition_transforms.go [hunk 1]
```diff
@@ -73,6 +73,29 @@ type MathTransform struct {
...
+	if m.Multiply == nil {
+		return nil, customError{err:"no multiplier provided for math transform"}
+	}
+	switch i := input.(type) {
+	case int64:
+		return float64(*m.Multiply) * float64(i), nil
+	case int:
+		return float64(*m.Multiply) * float64(i), nil
+	default:
+		return nil, customError{err:"input for math transform is not a number"}
```
Comment (ytsarev) [inline apis/apiextensions/v1/composition_transforms.go:95]: Please comply with the error reporting consistent with the rest of the codebase.  Creating a custom error for a single function looks suboptimal.

**Review by ytsarev — CHANGES_REQUESTED** [review]
Thanks for your contribution.

Please comply with DCO by signing the commit.

File: apis/apiextensions/v1/composition_transforms.go (hunk 1 above)
Comment (Vrashabh-Sontakke) [reply apis/apiextensions/v1/composition_transforms.go]: Thanks, I refered from other files and made those changes

https://github.com/Vrashabh-Sontakke/crossplane/blob/c9d66544c2fe3924036e76126618b9a94504eaf1/apis/apiextensions/v1/composition_transforms.go#L77-L95

File: apis/apiextensions/v1/composition_transforms.go [hunk 2]
```diff
@@ -73,6 +74,26 @@ type MathTransform struct {
...
+	switch i := input.(type) {
+	case int64:
+		return float64(*m.Multiply) * float64(i), nil
+	case int:
+		return float64(*m.Multiply) * float64(i), nil
+	default:
+		return nil, errors.New(errMathInputNonNumber)
+	}
+}
+
```
Comment (Vrashabh-Sontakke) [inline apis/apiextensions/v1/composition_transforms.go:96]: Thanks, I refered from other files and made those changes

File: apis/apiextensions/v1/composition_transforms.go [hunk 3]
```diff
@@ -73,6 +74,26 @@ type MathTransform struct {
...
+const (
+	errMathNoMultiplier   = "no multiplier provided for math transform"
+	errMathInputNonNumber = "input for math transform is not a number"
+)
+
+// Resolve runs the Math transform.
+func (m *MathTransform) Resolve(input interface{}) (interface{}, error) {
+	if m.Multiply == nil {
+		return nil, errors.New(errMathNoMultiplier)
+	}
```
Comment (Vrashabh-Sontakke) [inline apis/apiextensions/v1/composition_transforms.go:86]: Thanks, I refered from other files and made those changes

**Comment by ytsarev** [comment]
@Vrashabh-Sontakke, please also comply with the original PR template and provide the extensive testing report. Thank you.

**Comment by Vrashabh-Sontakke** [comment]
@ytsarev , sorry, I had to close the PR to follow the DCO in all commits and create a new one
https://github.com/crossplane/crossplane/pull/3669

**Pull request closed without merging**